Fix/17063/avoid undefined args

evont opened the pull request:
Issue: #17063

## What I did
fix args values not present in url are casted to !undefined for controls of type select and radio [issue](https://github.com/storybookjs/storybook/issues/17063) 

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

shilman committed:
6.4.2 latest.json version file

shilman committed:
Merge pull request #16850 from storybookjs/16847-do-not-hide-story

Preview: Don't hide the story while preparing

shilman committed:
Merge pull request #16853 from storybookjs/16768-modernInline-respect-no-inline

Don't render with `modernInline` if `inlineStories` is `false`

shilman committed:
6.4.3 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.3 [ci skip]

shilman committed:
v6.4.3

shilman committed:
Update git head to 6.4.3, update yarn.lock

shilman committed:
6.4.3 latest.json version file

shilman committed:
Merge pull request #16866 from storybookjs/16864-fix-mainjsframework-automigrate

CLI: Fix mainjsFramework automigrate

shilman committed:
6.4.4 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.4 [ci skip]

shilman committed:
v6.4.4

shilman committed:
Update git head to 6.4.4, update yarn.lock

shilman committed:
6.4.4 latest.json version file

shilman committed:
Merge pull request #16873 from storybookjs/11334-addon-docs-transclusion-fix

Addon-docs: Fix transclusion crash on webpack rules without test field

shilman committed:
Merge pull request #16882 from Marklb/marklb/fix-tsconfig-paths

Angular: Fix tsConfig paths not resolving for Angular >=12.2

shilman committed:
Fix typos

shilman committed:
Merge pull request #16878 from zarahzachz/patch-1

Fix typo.

shilman committed:
Merge pull request #16883 from storybookjs/16881-bump-lodash

Bump lodash to 4.17.21

shilman committed:
6.4.5 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.5 [ci skip]

shilman committed:
v6.4.5

shilman committed:
Update git head to 6.4.5, update yarn.lock

shilman committed:
6.4.5 latest.json version file

shilman committed:
Merge pull request #16885 from avergos29/fix-angular-cli-issue

Angular: Fix incorrect log

shilman committed:
6.4.6 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.6 [ci skip]

shilman committed:
v6.4.6

shilman committed:
Update git head to 6.4.6, update yarn.lock

shilman committed:
6.4.7 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.7 [ci skip]

shilman committed:
v6.4.7

shilman committed:
Update git head to 6.4.7, update yarn.lock

shilman committed:
6.4.7 latest.json version file

jonniebigodes committed:
Merge pull request #16901 from storybookjs/7902-fix-mdx-docs

Remove extra line breaks in MDX multi-story examples

shilman committed:
Merge pull request #16913 from storybookjs/16891-fix-autotitle-webpack5

Core: Fix auto-title in webpack5

shilman committed:
Merge pull request #16897 from storybookjs/16787-fix-clobber-vuerc

CLI: Fix `sb repro` clobbering .vuerc

shilman committed:
Merge pull request #16915 from storybookjs/16845-fix-hierarchy-separator-warning

Core: Add feature flag to disable legacy hierarchy separator warning

shilman committed:
6.4.8 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.8 [ci skip]

shilman committed:
v6.4.8

shilman committed:
Update git head to 6.4.8, update yarn.lock

shilman committed:
6.4.8 latest.json version file

shilman committed:
Merge pull request #16917 from storybookjs/16839-fix-duplicate-types

Core: Remove unused and occluded types

shilman committed:
Merge pull request #16919 from storybookjs/16862-fix-vue-rerender-viewMode

Core: Be explicit about `viewMode` to fix Vue issue

jonniebigodes committed:
Merge pull request #16927 from richorrichard/15631-change-theming-docs-wording-normal-to-light

Remove reference to "normal" theme, change to "light"

shilman committed:
Merge pull request #16931 from nagashimam/PR-#15147

Angular: Fix support for non-roman alphabets in story titles

shilman committed:
Merge pull request #16947 from storybookjs/16877-await-init-before-cache

Core: Ensure we have a full story index before caching

shilman committed:
6.4.9 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.9 [ci skip]

shilman committed:
v6.4.9

shilman committed:
Update git head to 6.4.9, update yarn.lock

shilman committed:
6.4.9 latest.json version file

shilman committed:
Merge pull request #16972 from storybookjs/docs/fix-mdx-whitespace

Fix whitespace bugs in MDX examples

shilman committed:
Merge pull request #16980 from storybookjs/docs/angular-6-4-migration

Added MIGRATION instructions for Angular 13 / SB6.4 for Angular

shilman committed:
Merge pull request #16983 from storybookjs/docs/fix-a11y-readme

Fix a11y readme

shilman committed:
Merge pull request #16988 from virtuoushub/fix/debug-output-for-ci

Core: Fix debug output on webpack failures

shilman committed:
Merge pull request #16989 from WIVSW/button-story-snippets-for-html

Button story snippets for html framework

shilman committed:
Merge pull request #17027 from mgcrea/patch-source-loader

Source-loader: Fix node.declaration edge case

shilman committed:
Merge pull request #17032 from mandarini/fix/angular-131-method-rename

Angular: Fix for renamed method in angular 13.1

MichaelArestad committed:
Merge pull request #17009 from silversonicaxel/fix-jest-docs

Set `string` to `string[]` when  is not necessary in jest addon documentation

jonniebigodes committed:
Merge pull request #17061 from drakang4/patch-1

fix typo in MIGRATION.md

jonniebigodes committed:
Merge pull request #17074 from storybookjs/docs-sharing

Re-arrange docs for the new /sharing section

jonniebigodes committed:
Merge pull request #17084 from storybookjs/docs-sharing

Minor docs bugs

jonniebigodes committed:
Merge pull request #17085 from storybookjs/docs-aside-note-tweak

Docs: Don't print "Note" in <aside>s

shilman committed:
Merge pull request #16728 from Marklb/marklb/fix-run-compodoc-on-windows

Angular: Fix runCompodoc for Windows, local Compodoc, and user specified tsconfig

kylegach committed:
Merge pull request #17088 from orrgottlieb/patch-1

Vibe Design System

shilman committed:
Merge pull request #17096 from cameliaben/patch-1

Fix typo in story-rendering.md

shilman committed:
Merge pull request #17037 from WIVSW/button-story-with-args-snippets-for-html

Button story with args snippets for html framework

shilman committed:
Merge pull request #17022 from Taillook/chore/react-dev-utils

React: Remove react-dev-utils

shilman committed:
Merge pull request #17046 from iagopiimenta/patch-1

Add Reactstrap UI

shilman committed:
Merge pull request #17134 from storybookjs/fix/16918-mdx-hash-link

Router: Fix navigating to hash links

shilman committed:
Merge pull request #17131 from storybookjs/angular/fix-angular-13.1

Angular: Fix angular 13.1 JIT error and HMR reload

jonniebigodes committed:
Merge pull request #17149 from storybookjs/chore_minor_fix_configure_docs

Chore: (Docs) removes incorrect info in Configure section

shilman committed:
Merge pull request #17153 from storybookjs/chore_minor_api_docs_changes

Chore: (Docs) Updates the api/CSFsection

shilman committed:
Merge pull request #17124 from storybookjs/chore_fix_toc

Chore: (Docs) Updates to the sidebar toc file

shilman committed:
Merge pull request #17156 from storybookjs/angular/fix-angular-13.1

Angular: Use ɵReflectionCapabilities to find component & module metadata

jonniebigodes committed:
Merge pull request #17125 from storybookjs/chore_remove_notes_usage

Chore: (Docs) Removes missing note from asides

jonniebigodes committed:
Merge pull request #17123 from storybookjs/chore_docs_minor_fixes_get_started

Chore: (Docs) Minor tweaks to the install section

shilman committed:
Merge pull request #17180 from storybookjs/17179-fix-cli-table3-colors-dep

Bump cli-table3 to fix colors bug

shilman committed:
Merge pull request #17174 from storybookjs/17128-fix-process-env-assignment

Core: Fix process.env assignment

shilman committed:
6.4.10 changelog

shilman committed:
Update CHANGELOG.md

shilman committed:
Update root, peer deps, version.ts/json to 6.4.10 [ci skip]

shilman committed:
v6.4.10

shilman committed:
Update git head to 6.4.10, update yarn.lock

shilman committed:
6.4.10 latest.json version file

shilman committed:
Merge pull request #17185 from storybookjs/autotitle-path

CSF3: Remove `path` from autoTitle browser code

shilman committed:
Merge pull request #17173 from friederbluemle/patch-1

Fix TypeScript spelling

shilman committed:
Merge pull request #17119 from inokawa/patch-1

Fix link in readme of @storybook/addon-storysource

shilman committed:
6.4.11 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.11 [ci skip]

shilman committed:
v6.4.11

shilman committed:
Update git head to 6.4.11, update yarn.lock

shilman committed:
6.4.12 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.12 [ci skip]

shilman committed:
v6.4.12

shilman committed:
Update git head to 6.4.12, update yarn.lock

shilman committed:
6.4.12 latest.json version file

jonniebigodes committed:
Merge pull request #17203 from storybookjs/chore_fix_repro_docs

Chore: Updates the how to reproduce docs

jonniebigodes committed:
Merge pull request #17208 from storybookjs/chore_docs_updates_syntax_highlight_docs

Chore: (Docs) Minor updates to the syntax highlight documentation.

shilman committed:
Merge pull request #17213 from storybookjs/16067-fix-manager-process

Webpack5: Fix manager.js process references

shilman committed:
Merge pull request #17206 from storybookjs/angular/fix-angular-13.1

Angular: Fix 13.1 and add CI test cases

jonniebigodes committed:
Merge pull request #17221 from storybookjs/chore_docs_fix_addon_knowledge_base

Chore: (Docs) Minor tweaks to the addon knowledge base

jonniebigodes committed:
Merge pull request #17224 from storybookjs/chore_docs_update_addons_install_docs

Chore: (Docs) Minor adjustments to the install addons docs

shilman committed:
Merge pull request #17240 from storybookjs/15574-fix-namedexportsorder-warning

Core: Fix __namedExportsOrder warning from preview.js

shilman committed:
Merge pull request #17239 from storybookjs/16820-fix-prettier-transpilation

Fix prettier transpile problems

shilman committed:
Merge pull request #17241 from storybookjs/17008-fix-staticdirs-favicon

Core: Fix staticDirs favicon handling by refactor

jonniebigodes committed:
Merge pull request #17244 from storybookjs/chore_docs_fix_composition_link

Chore: (Docs) Fix dead link in sharing section

shilman committed:
Fix prettier 2.3 formatting on main

shilman committed:
Update snapshots

shilman committed:
Merge pull request #17245 from storybookjs/fix-prettier-2-3-formatting-main

Fix prettier 2.3 formatting on main

shilman committed:
6.4.13 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.13 [ci skip]

shilman committed:
v6.4.13

shilman committed:
Update git head to 6.4.13, update yarn.lock

shilman committed:
6.4.13 latest.json version file

jonniebigodes committed:
Merge pull request #17248 from storybookjs/chore_docs_expand_api_docs

Chore: (Docs) Complements the Addon API docs

shilman committed:
Merge pull request #17249 from merceyz/merceyz/fix/process-fallback

Webpack: Fix for `process` fallback using `require.resolve`

shilman committed:
Merge pull request #17252 from storybookjs/17038-16964-16924-fix-complex-globs

Core: Fix issue with recursive glob with prior special chars

shilman committed:
Merge pull request #17256 from storybookjs/16895-fix-docs-disable

Addon-docs: Fix `docs.disable` parameter on DocsPage

shilman committed:
Merge pull request #17257 from storybookjs/17255-fix-prettier-parser-babel

Restore prettier >= 2.2.1 to satisfy previous constraints

shilman committed:
Merge pull request #17265 from storybookjs/17261-fix-jscodeshift-colorsjs-dep

Upgrade jscodeshift dependency for colors.js bug

shilman committed:
Merge pull request #17266 from storybookjs/upgrade-compodoc-colors-dep

Upgrade compodoc for colors.js bug

jonniebigodes committed:
Merge pull request #17293 from ankon/patch-1

docs: Point to the measure-and-outline page

shilman committed:
Skip version check in CI

shilman committed:
Merge pull request #17300 from storybookjs/15933-build-storybook-no-manager-cache

CLI: Add `--no-manager-cache` to build-storybook

shilman committed:
Merge pull request #17286 from rasmustnilsson/patch-1

Fix example syntax by adding missing `=`

shilman committed:
6.4.14 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.14 [ci skip]

shilman committed:
v6.4.14

shilman committed:
Update git head to 6.4.14, update yarn.lock

shilman committed:
6.4.14 latest.json version file

jonniebigodes committed:
Merge pull request #17114 from cabbiepete/patch-1

Gotcha about chromatic publish not working

ndelangen committed:
Merge pull request #17318 from storybookjs/docs/update-migration-cra-preset-webpack5

docs(MIGRATION): add CRA preset disclaimer in Webpack 5 section

shilman committed:
Upgrade to yarn 3.1.1

jonniebigodes committed:
Merge pull request #17359 from storybookjs/chore_docs_updates_interactions_docs

Chore: (Docs) Updates the docs to address #17166

shilman committed:
Merge pull request #17356 from storybookjs/cli-add-package-version-parameter

CLI: Add version update argument to generate-sb-packages-versions utility

shilman committed:
Merge pull request #17328 from clentfort/fix/16964-negated-globs

Core: Fix negated glob support

shilman committed:
Fix snapshot

shilman committed:
Merge pull request #17323 from storybookjs/upgrade-main-yarn

Build: Upgrade main yarn

shilman committed:
Merge branch 'main' of github.com:storybookjs/storybook into main

shilman committed:
6.4.15 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.15 [ci skip]

shilman committed:
v6.4.15

shilman committed:
Update git head to 6.4.15, update yarn.lock

shilman committed:
6.4.15 latest.json version file

shilman committed:
Merge pull request #17363 from matheo/angular/fix-nx

Angular: Retrieve version from core package

shilman committed:
Merge pull request #17334 from storybookjs/fix-compodoc-for-windows

Angular: Workaround for compodoc on windows machines

shilman committed:
6.4.16 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.16 [ci skip]

shilman committed:
v6.4.16

shilman committed:
Update git head to 6.4.16, update yarn.lock

shilman committed:
6.4.16 latest.json version file

shilman committed:
Merge pull request #17372 from storybookjs/revert-17363-angular/fix-nx

Revert "Angular: Retrieve version from core package"

shilman committed:
6.4.17 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.17 [ci skip]

shilman committed:
v6.4.17

shilman committed:
Update git head to 6.4.17, update yarn.lock

shilman committed:
6.4.17 latest.json version file

jonniebigodes committed:
Merge pull request #17378 from jonbeller/patch-4

Update login-form-with-play-function.js.mdx

shilman committed:
Merge pull request #17376 from jonbeller/patch-1

Update interaction-testing.md

shilman committed:
Merge pull request #17395 from storybookjs/fix/mdxv2

CLI: Pin version of `@mdx-js/react` to 1.x.x until we are compatible

shilman committed:
6.4.18 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.18 [ci skip]

shilman committed:
v6.4.18

shilman committed:
Update git head to 6.4.18, update yarn.lock

shilman committed:
6.4.18 latest.json version file

jonniebigodes committed:
Merge pull request #17403 from storybookjs/chore_docs_updates_decorators

Chore: (Docs) Updates decorators docs

jonniebigodes committed:
Merge pull request #17411 from storybookjs/chore_update_essentials_overview

Chore: (Docs) Minor updates to the Essentials introduction

shilman committed:
Merge pull request #17415 from flyingSquirrel-dev/patch-1

Fix typo in presets docs

shilman committed:
Merge pull request #17345 from storybookjs/feat/add-interactions-to-cli-template

CLI/React: Add interactions to cli template

shilman committed:
Merge pull request #17404 from symmetriq/section-row-fix-typo

Addon-docs: Fix typo in ArgsTable tooltip

shilman committed:
Merge pull request #17401 from storybookjs/fix_addon_support_table

Chore: (Docs) Removes outdated support table and updates links

shilman committed:
Merge pull request #17435 from storybookjs/fix-configure-error-handling

Core: Ensure we show an error when `configure()` throws

shilman committed:
Merge pull request #17387 from storybookjs/chore_docs_updates_for_angular

Chore: (Docs) Updates to the Angular documentation

shilman committed:
Merge pull request #17351 from yhor1e/patch-1

Fix jest-image-snapshot link

shilman committed:
Merge pull request #17327 from evancharlton/useParameter-fallback

Core: Fix `useParameter` with nullish coalescing

shilman committed:
Merge pull request #17434 from huyenltnguyen/fix/link-to

Addon-links: Fix export statement in react.d.ts

shilman committed:
Merge pull request #17158 from fraserisland/patch-1

Update login-form-with-play-function.mdx.mdx

jonniebigodes committed:
Merge pull request #17440 from storybookjs/chore_docs_minor_tweaks_writing_stories_intro

Chore: (Docs) Minor updates to writing writing stories intro

shilman committed:
Merge pull request #17266 from storybookjs/upgrade-compodoc-colors-dep

Upgrade compodoc for colors.js bug

kylegach committed:
Remove /docs/get-started/examples page

- Will redirect to /showcase in corresponding frontpage PR
    - https://github.com/storybookjs/frontpage/pull/334
- Add examples link to how-to-contribute page

shilman committed:
Merge pull request #17447 from storybookjs/add-preview-extract

Core/CLI: Add `extract` function to `PreviewWeb` and use it in `sb extract` if available

kylegach committed:
Merge pull request #17464 from storybookjs/remove-docs-examples-page-2

Remove /docs/get-started/examples page

jonniebigodes committed:
Merge pull request #17484 from nam-hle/patch-4

docs: The Template is a HOC

shilman committed:
Merge pull request #17358 from storybookjs/chore_docs_fix_storybook_config_overview

Chore: (Docs) Updates Storybook configuration docs for TS and flags.

shilman committed:
6.4.19 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.19 [ci skip]

shilman committed:
v6.4.19

shilman committed:
Update git head to 6.4.19, update yarn.lock

shilman committed:
6.4.19 latest.json version file

jonniebigodes committed:
Merge pull request #17479 from DovydasNavickas/patch-1

Double property initialization.

kylegach committed:
Merge pull request #17316 from WIVSW/your-component-snippets-for-html

Your component snippets for html

jonniebigodes committed:
Merge pull request #17511 from storybookjs/chore_docs_remove_misleading_info_doc_blocks

Chore: (Docs) Removes misleading info from documentation

shilman committed:
Merge pull request #17532 from storybookjs/shilman-patch-2

Fix incorrect docs on buildStoriesJson

jonniebigodes committed:
Merge pull request #17531 from oligafner/add-a-good-old-web-component-snippet

add list-story-unchecked snippet

jonniebigodes committed:
Merge pull request #17554 from fbredius/docs/add-snippet-for-preact

Docs: Add Preact code snippet for Whats a Story

jonniebigodes committed:
Merge pull request #17546 from storybookjs/e2e-docs

Add brief guide to running e2e tests

jonniebigodes committed:
Merge pull request #17551 from storybookjs/chore_docs_doc_blocks_updates

Chore: (Docs)  Doc Blocks updates

kylegach committed:
Simplify community frameworks

- Remove marko, mithril, rax, & riot

shilman committed:
Merge pull request #17582 from storybookjs/fix_writing_docs_aside

Chore: (Docs) Updates to the Writing Docs asides

jonniebigodes committed:
Merge pull request #17612 from MrAnyx/patch-1

Fix typo in the sub component documentation

shilman committed:
Merge pull request #17606 from yanick/patch-3

Core: Fix core.builder check

shilman committed:
Merge pull request #17659 from storybookjs/fix/sorted-exports-bug

shilman committed:
Merge pull request #17669 from storybookjs/17611-docs-order-should-mirror-index

Addon-docs: DocsPage story order should match the index

jonniebigodes committed:
Merge pull request #17520 from juri-sinitson/patch-3

Add documentation for Nx 13 with Angular 13 #17419

jonniebigodes committed:
Merge pull request #17689 from JanMarkD/fix-spelling-testing-docs

Docs: Fix minor typo in Importing Stories in Tests documentation

jonniebigodes committed:
Merge pull request #17690 from storybookjs/chore_docs_minor_controls_polish

Chore: (Docs) Additional polish to the Controls documentation

jonniebigodes committed:
Merge pull request #17691 from storybookjs/chore_faq_updates

Chore: (Docs) Minor addition to the faq

shilman committed:
Merge pull request #17702 from tacc-wbomar/patch-1

Web-components: Fix CSS class usage in CLI template

shilman committed:
Merge pull request #17687 from Jarvis1010/patch-1

make command for html explicit

shilman committed:
Merge pull request #17679 from storybookjs/norbert/ch-1636-storybook-composition-ref-is-shown-when

UI: Fix composition support in safari

jonniebigodes committed:
Merge pull request #17716 from ch3coo2ca/fix-typo-in-react-snippets

fix : typo in props name

kylegach committed:
Merge pull request #17658 from fbredius/preact-code-snippet

Docs: Add remaining Preact code snippets for Get-Started section

jonniebigodes committed:
Merge pull request #17736 from storybookjs/fix_preact_snippets

Chore: (Docs) Fix typo in preact snippets

shilman committed:
Merge pull request #17734 from storybookjs/chore_docs_env_vars_updates

Chore: (Docs) Updates the env vars docs for sb env field

shilman committed:
Merge pull request #17729 from ch3coo2ca/fix-missing-await

fix : add missing await for an async function

jonniebigodes committed:
Merge pull request #17776 from storybookjs/chore_docs_fix_html_snippets

Chore: (Docs) Fixes the incorrect path in the snippets

jonniebigodes committed:
Merge pull request #17718 from JodiWarren/patch-1

Fix typo in docs

shilman committed:
Merge pull request #17633 from storybookjs/angular/fix-setter

Angular: Fix multiple calls of Input setter

shilman committed:
Merge pull request #17829 from storybookjs/feat/add-vite-builder-automigration

CLI: Add automigration to `@storybook/builder-vite`

shilman committed:
Merge pull request #17830 from storybookjs/feat/cli-add-vite-builder

CLI: Fix vite/jest issue with mocked global

shilman committed:
Merge pull request #17833 from inclusive-coding-bot/switch-to-inclusive-terms

Switch to gender neutral terms

shilman committed:
6.4.20 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.20 [ci skip]

shilman committed:
v6.4.20

shilman committed:
Update git head to 6.4.20, update yarn.lock

shilman committed:
6.4.20 latest.json version file

jonniebigodes committed:
Merge pull request #17854 from hiro0218/fix/typo-docs

Fix typo docs files

shilman committed:
Merge pull request #17858 from storybookjs/tech/automigrate-preserve-quotes

CLI: Preserve quote style in automigrate

kylegach committed:
Merge pull request #17780 from fbredius/17225-date-control

Controls: Fix date control width in addons panel

shilman committed:
Merge pull request #17876 from Coly010/fix-angular-builder-for-nx

Angular: Do not use default for includePaths

shilman committed:
Merge pull request #17909 from storybookjs/fix/cli-upgrade-exclude-list

CLI: Update the exclude list for upgrade warnings

jonniebigodes committed:
Merge pull request #17758 from Pepijnk12/documentation-fixes

Fix documentation typos and grammatical errors

shilman committed:
6.4.21 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.21 [ci skip]

shilman committed:
v6.4.21

shilman committed:
Update git head to 6.4.21, update yarn.lock

shilman committed:
6.4.21 latest.json version file

jonniebigodes committed:
Merge pull request #17955 from oliviavoler/patch-3

Fix typo in Annotation section

shilman committed:
Merge pull request #17875 from storybookjs/core-client

Core: Avoid framework imports from core/client

shilman committed:
6.4.22 changelog

shilman committed:
Update root, peer deps, version.ts/json to 6.4.22 [ci skip]

shilman committed:
v6.4.22

shilman committed:
Update git head to 6.4.22, update yarn.lock

shilman committed:
6.4.22 latest.json version file

Commit:
fix(store): not set undefined to acc when key is not exist in args

Commit:
fix(store): avoid get into warning

kylegach reviewed (commented):
Thank for fixing this important issue, @evont! And with your very first contribution. 👏 

@shilman @tmeasday — This change seems sound to me, but the args code in general is likely quite nuanced and I think it would benefit from you taking a look as well. Also, is it OK that this PR is against `main` instead of `next`?

kylegach commented:
@evont — I'd do this myself, but I cannot push to your fork's branch. Can you please add this story, to the `examples/official-storybook/stories/core/args.stories.js` file?

```js
export const TestUndefinedArgs = Template.bind({});
TestUndefinedArgs.args = {
  first: 'Bob',
  last: 'Miller',
  foo: 'bar',
};
TestUndefinedArgs.argTypes = {
  first: {
    options: ['Bob', 'Alice'],
  },
  last: {
    options: ['Miller', 'Meyer'],
  },
  foo: {
    control: { type: 'text' },
  },
};
```

That will allow maintainers to easily test your change (use the same repro steps as the linked issue) in the deployed Storybook. Thanks!

shilman reviewed (changes requested):
Hi @evont @kylegach -- thanks for the fix! Indeed all PRs should go against `next` not `main` unless it's a special case, which AFAICT this is not. Please rebase & retarget the PR. 🙏

tmeasday commented on lib/store/src/args.ts:
```diff
@@ -101,7 +101,8 @@ export const validateOptions = (args: Args, argTypes: ArgTypes): Args => {
     const isValidArray = isArray && invalidIndex === -1;
 
     if (args[key] === undefined || options.includes(args[key]) || isValidArray) {
```
This check for `args[key] === undefined` seems a little pointless if we later do `if (key in args)`. Do we understand why the code was there in the first place? We should get rid of it, and perhaps figure that out.

tmeasday replied:
Or does it remain here to avoid the warning lower down?

tmeasday reviewed (changes requested):
Thanks for the contribution @evont. Can you please add test case(s) that demonstrate the original problem?

Commit:
add arg story to test change

evont commented:
> @evont — I'd do this myself, but I cannot push to your fork's branch. Can you please add this story, to the `examples/official-storybook/stories/core/args.stories.js` file?
> 
> ```js
> export const TestUndefinedArgs = Template.bind({});
> TestUndefinedArgs.args = {
>   first: 'Bob',
>   last: 'Miller',
>   foo: 'bar',
> };
> TestUndefinedArgs.argTypes = {
>   first: {
>     options: ['Bob', 'Alice'],
>   },
>   last: {
>     options: ['Miller', 'Meyer'],
>   },
>   foo: {
>     control: { type: 'text' },
>   },
> };
> ```
> 
> That will allow maintainers to easily test your change (use the same repro steps as the linked issue) in the [deployed Storybook](https://deploy-preview-17973--storybookjs.netlify.app/official-storybook/?path=/story/core-args--test-undefined-args). Thanks!

Sure, with pleasure, I have added it and push it to the branch, except I added `control: { type: 'select' }` because it is odd that if I don't do that, the `options` will not transform into `select` but `text`

evont commented:
Sorry for my mistake, fork from main branch and rebase to next will be that tough, so I decided to fork form next and make a new PR, please check in this [new PR](https://github.com/storybookjs/storybook/pull/18135)
@tmeasday @kylegach @shilman

shilman commented:
Thanks @evont! Closing this in favor of #18135
